lib/auth/oidc.go

```diff
@@ -657,8 +772,12 @@ func mergeClaims(a jose.Claims, b jose.Claims) (jose.Claims, error) {
 
 // getClaims gets claims from ID token and UserInfo and returns UserInfo claims merged into ID token claims.
 func (a *Server) getClaims(oidcClient *oidc.Client, connector types.OIDCConnector, code string) (jose.Claims, error) {
+	return getClaimsFun(a.closeCtx, oidcClient, connector, code)
+}
 
-	oac, err := a.getOAuthClient(oidcClient, connector)
+// getClaimsFun implements Server.getClaims, but allows that code path to be overridden for testing.
+var getClaimsFun = func(closeCtx context.Context, oidcClient *oidc.Client, connector types.OIDCConnector, code string) (jose.Claims, error) {
```

Comment:
Can we move the `getClaimsFun` to the `Server` settings instead of using global variable ?

Reply:
TBH I'm not sure which is cleaner -- note that `getClaimsFun` is not exported, so not a true global -- but it being in settings for `Server` is your preference, I can make it happen.